types/tx/msgs_test.go

```diff
@@ -0,0 +1,195 @@
+package tx_test
+
+import (
+	"testing"
+
+	"github.com/cosmos/gogoproto/proto"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+)
+
+func Test_SetMsg(t *testing.T) {
+	cases := []struct {
+		name   string
+		msg    sdk.Msg
+		expErr bool
+	}{
+		{
+			name:   "Set nil Msg",
+			msg:    nil,
+			expErr: true,
+		},
+		{
+			name:   "Set a valid message",
+			msg:    &DummyProtoMessage1{},
+			expErr: false,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, err := tx.SetMsg(tc.msg)
+			if tc.expErr {
+				require.Error(t, err)
+				return
+			}
+
+			expected := mustAny(tc.msg)
```

Comment:
This calls the same `types.NewAnyWithValue` method as `tx.SetMsg`. If you want to ensure that the Any type has the expected data then you can verify cache, url and bytes. Or Unmarshal with and without cache

Reply:
Agree. It will be better to test cache, url and bytes. Thank you for the suggestion

Fixed